src/common/post.ts

```diff
@@ -621,3 +625,18 @@ export const findPostImageFromContent = ({
 
   return imgTag?.attrGet('src') || undefined;
 };
+
+export const getPostTranslatedTitle = (
+  post: Partial<Pick<Post, 'title' | 'contentMeta'>>,
+  contentLanguage: ContentLanguage,
+) =>
+  post.contentMeta?.translate_title?.translations?.[contentLanguage] ||
+  (post.title as string);
+
+export const getPostSmartTitle = (
+  post: Partial<Pick<Post, 'title' | 'contentMeta'>>,
+  contentLanguage: ContentLanguage,
+) =>
+  post.contentMeta?.alt_title?.translations?.[contentLanguage] ||
+  post.contentMeta?.alt_title?.translations?.[ContentLanguage.English] ||
+  getPostTranslatedTitle(post, contentLanguage);
```

Comment:
Added future support for when/if we support smart titles in other languages. It first tries specific language, then english, and finally fallbacks to original title.